pacai/agents/ghost/directional.py

```diff
@@ -47,7 +48,11 @@ def getDistribution(self, state):
             dist[a] = float(bestProb) / len(bestActions)
 
         for a in legalActions:
-            dist[a] += float(1 - bestProb) / len(legalActions)
-
-        dist.normalize()
+            temp_dist = float(1 - bestProb) / len(legalActions)
+            if not a in dist:
+                dist[a] = temp_dist
+            else:
+                dist[a] += temp_dist
```

Comment:
No, your still missing it a bit.

The pattern we want to use is checking for a dict key, and putting in the default value only if it doesn't exist.
This way we can deal with the special case (of the first appearance of a key) while still only including the code that does the real computations in a single place.
For example:
```
if (key not in dict):
    dict[key] = defaultValue
dict[key] += some computation
```